File: front/src/types/general.ts
export type GeneralConfig = {
    language: "ja-JP";
    soundEffect: boolean;
    characterOutputChange: boolean;
    transcription: {
        type: "openai_speech_to_text"|"google_speech_to_text"|"vosk_server"|"speech_recognition"
        method: "auto"|"pushToTalk"
        autoSetting: {
            threshold: number;
            silentThreshold: number;
            audioMinLength: number;
        };
    };
};

File: server/data/general.go
package data

type GeneralConfig struct {
	Background            string `json:"background"`
	Language              string `json:"language"`
	SoundEffect           bool   `json:"soundEffect"`
	CharacterOutputChange bool   `json:"characterOutputChange"`
	Transcription         struct {
		Type        string `json:"type"`
		Method      string `json:"method"`
		AutoSetting struct {
			Threshold       float64 `json:"threshold"`
			SilentThreshold float64 `json:"silentThreshold"`
			AudioMinLength  float64 `json:"audioMinLength"`
		} `json:"autoSetting"`
	} `json:"transcription"`
}
